infra/common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/dialect/OracleSchemaMetaDataLoader.java

```diff
@@ -74,7 +75,7 @@ public final class OracleSchemaMetaDataLoader implements DialectSchemaMetaDataLo
     public Collection<SchemaMetaData> load(final DataSource dataSource, final Collection<String> tables, final String defaultSchemaName) throws SQLException {
         Map<String, Collection<ColumnMetaData>> columnMetaDataMap = new HashMap<>(tables.size(), 1);
         Map<String, Collection<IndexMetaData>> indexMetaDataMap = new HashMap<>(tables.size(), 1);
-        try (Connection connection = dataSource.getConnection()) {
+        try (Connection connection = new MetaDataLoaderConnectionAdapter(DatabaseTypeFactory.getInstance("Oracle"), dataSource.getConnection())) {
```

Comment:
Why add MetaDataLoaderConnectionAdapter?

Reply:
Using ojdbc6 with defined development circumstance, there will be a NoSuchMethodError like "java.lang.AbstractMethodError: oracle.jdbc.driver.T4CConnection.getSchema()Ljava/lang/String;" when invoke following statement in OracleSchemaMetaDataLoader
```
    private Map<String, Collection<ColumnMetaData>> loadColumnMetaDataMap(final Connection connection, final Collection<String> tables) throws SQLException {
        ...
            preparedStatement.setString(1, connection.getSchema());
        ...
    }
```
With debug with connection.getSchema(). We could know ojdbc6 doesn't define method T4CConnection.getSchema().
To invoke connection.getSchema() smoothly, adapt a class like MetaDataLoaderConnectionAdapter